ui/src/app/utils/errors.spec.tsx

```diff
@@ -0,0 +1,144 @@
+import {
+  ErrorCode,
+  ErrorResponse,
+  WorkspaceUserRolesResponse,
+} from 'generated/fetch';
+
+import {
+  defaultErrorResponseFormatter,
+  errorHandlerWithFallback,
+  FALLBACK_ERROR_TITLE,
+} from './errors';
+import { NotificationStore } from './stores';
+
+describe('defaultErrorResponseFormatter', () => {
+  test.each([
+    ['the error response has no details', {}, 'An API error occurred.'],
+    [
+      'the error response could not be parsed, and no status code is present',
+      { errorCode: ErrorCode.PARSEERROR },
+      'An API error occurred.',
+    ],
+    [
+      'the error response could not be parsed, but a status code is present',
```

Comment:
when there's a parse error, statusCode is the only other field that can be set.  Not sure why.